docstr_coverage/printers.py

```diff
@@ -22,128 +25,475 @@
-        if self.verbosity >= 2:
-            self._print_file_statistics(results)
-        if self.verbosity >= 1:
-            self._print_overall_statistics(results)
+@dataclass(frozen=True)
+class OverallCoverageStat:
+    """Data Structure of coverage statistic."""
+
+    found: int
+    grade: str
+    is_skip_class_def: bool
+    is_skip_file_docstring: bool
+    is_skip_init: bool
+    is_skip_magic: bool
+    is_skip_private: bool
+    missing: int
+    needed: int
+    num_empty_files: int
+    num_files: int
+    total_coverage: float
+
 
-    def _print_file_statistics(self, results):
-        """Prints the file specific information to stdout.
+class Printer(ABC):
+    """Base abstract superclass for printing coverage results.
 
+    It provides coverage results in data structures (`OverallCoverageStat`, `FileCoverageStat` and
+    `IgnoredNode`) and abstract methods for implementing type of displaying and saving in file
+    statistic data.
+
+    In heir classes you can use `overall_coverage_stat` and `overall_files_coverage_stat`
+    attributes. Depends of given `verbosity` some data can be `None`."""
+
+    def __init__(
+        self,
+        results: ResultCollection,
+        verbosity: int,
+        ignore_config: IgnoreConfig = IgnoreConfig(),
+    ):
+        """
         Parameters
         ----------
         results: ResultCollection
-            The information about docstr presence to be printed to stdout."""
-        for file_path, file in results.files():
-            if self.verbosity < 4 and file.count_aggregate().missing == 0:
-                # Don't print fully documented files
-                continue
-
-            # File Header
-            print_line('\nFile: "{}"'.format(file_path))
-
-            # List of missing docstrings
-            if self.verbosity >= 3:
-                if file.status == FileStatus.EMPTY and self.verbosity > 3:
-                    print_line(" - File is empty")
-                for expected_docstr in file._expected_docstrings:
-                    if expected_docstr.has_docstring and self.verbosity > 3:
-                        print_line(
-                            " - Found docstring for `{0}`".format(expected_docstr.node_identifier)
-                        )
-                    elif expected_docstr.ignore_reason and self.verbosity > 3:
-                        print_line(
-                            " - Ignored `{0}`: reason: `{1}`".format(
-                                expected_docstr.node_identifier, expected_docstr.ignore_reason
-                            )
+            Coverage analyze results.
+        verbosity: int
+            Verbosity identifier.
+        ignore_config: IgnoreConfig
+            Config with ignoring setups.
+        """
+        self.verbosity = verbosity
+        self.ignore_config = ignore_config
+        self.results = results
+        self.__overall_coverage_stat = None
```

Comment:
type hints would be nice (same for next field, or, for consistency, for all fields)

Reply:
Add more type hints in the next update. I think I have worked too much on it :) 
Please have a look at it, I think we can remove a couple of hints.